### ISSUE TITLE
Is it possible to scale back the monkey_patching of a module on Windows?

### ISSUE THREAD

**Issue by MPagel** [eventlet/eventlet#386]
When attempting to `monkey_patch` the `OS` module (before an `import tempfile` statement), I get

> NotImplementedError: set_nonblocking() on a file object with no setblocking() method (Windows pipes don't support non-blocking I/O) 

I get this error if the `os` module is monkey_patched and I'm attempting an `obj.write()` of a temp file (created by a call of `obj = tempfile.SpooledTemporaryFile(...)`)

Should I only `monkey_patch` after importing the `tempfile` module, so that module is excluded? Can I partially patch `os` so that file objects are not "greened"?

**Comment by temoto** [comment]
TL;DR: no, yes.

You can use `eventlet.patcher.original()` to get unpatched version of module.

You can select which modules to include or exclude from patching via kwargs to `monkey_patch()`. Read http://eventlet.net/doc/patching.html#eventlet.patcher.monkey_patch It contains a small heuristic to determine whether you wanted "include-only-these" or "all-excluding-these" which should be intuitive to use.

It is very recommended to run `monkey_patch()` in exactly first line of your program, before anything else. Although it's not always possible, so patcher makes sensible attempt to fix previously imported code. You should expect `import tempfile ; then monkey_patch()` work the same as in reverse order. Consider opposite a bug that may be fixed in future releases.

Following code is Undefined Behavior. It may work, it may publish your database on twitter, it may erase your disks or launch nuclear rockets.
```
try:
  eventlet.monkey_patch()
except NotImplementedError:
  pass
```

**Comment by temoto** [comment]
Please reopen this issue if you feel so.
